idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -0,0 +1,507 @@
+        def psi_s_eqn(b, t):
+            return b.psi_s[t] == 2 * b.ys_model[t] / (b.U2[t]**2)
+
+    def initialize(self, state_args={}, outlvl=idaeslog.NOTSET, solver='ipopt',
+                   optarg={'tol': 1e-6, 'max_iter': 50}):
+        """
+        Initialize the inlet compressor stage model.
+        This deactivates the specialized constraints,
+        then does the isentropic compressor initialization,
+        then reactivates the constraints and solves.
+
+        Args:
+            state_args (dict): Initial state for property initialization
+            outlvl: Initialization logger level
+            solver (str): Solver to use for initialization
+            optarg (dict): Solver arguments dictionary
+        """
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(self.name, outlvl, tag="unit")
+
+        opt = pyo.SolverFactory(solver)
+        opt.options = optarg
+
+        unfix_ratioP = {}
+        for t in self.flowsheet().config.time:
+            # if there is not a good guess for efficiency or outlet pressure
+            # provide something reasonable.
+            eff = self.efficiency_isentropic[t]
+            eff.fix(
+                eff.value if pyo.value(eff) > 0.3 and pyo.value(
+                    eff) < 1.0 else 0.85)
+            # check for alternate pressure specs
+            if self.outlet.pressure[t].fixed:
+                self.ratioP[t] = pyo.value(
+                    self.outlet.pressure[t] / self.inlet.pressure[t])
+            elif self.control_volume.deltaP[t].fixed:
+                self.ratioP[t] = pyo.value(
+                    (self.control_volume.deltaP[t] + self.inlet.pressure[t]
+                     ) / self.inlet.pressure[t])
+            elif self.ratioP[t].fixed:
+                self.outlet.pressure[t] = pyo.value(
+                    self.ratioP[t] * self.inlet.pressure[t])
+            else:
+                if value(self.ratioP[t]) is None or value(
+                        self.ratioP[t]) < 1.01:
+                    self.ratioP[t].fix(1.5)
+                else:
+                    self.ratioP[t].fix()
+                unfix_ratioP[t] = True
+
+        # set list of constraints
+        constraint_list = [
+            "impeller_work_coeff_eqn",
+            "polytropic_head_coeff_vaned_diffuser_eqn",
+            "polytropic_head_coeff_eqn",
+            "polytropic_efficiency_eqn",
+            "Ma_con",
+            "rspeed_con",
+            "eff_p_v_cons",
+            "polytropic_correlation",
+            "delta_enth_polytropic_con",
+            "mass_flow_coeff_eqn",
+            "psi_3_eqn",
+            "psi_s_eqn"]
+
+        # deactivate unit model level constraints
+        for c in self.component_objects(pyo.Constraint, descend_into=True):
+            if c.local_name in constraint_list:
+                c.deactivate()
```

Comment:
This is kind of a nit pick, but a more straight-forward way would be

```python

for c in constraint_list:
    self.component(c).deactivate()
```
I guess that's assuming the constraints are attached to ```self```, which they probably are but I didn't check.